edi_voxel_account_invoice_oca/reports/report_voxel_invoice.py

```diff
@@ -0,0 +1,184 @@
+            "customers": self._get_customers_data(docs),
+            "comments": self._get_comments_data(docs),
+            "references": self._get_references_data(docs),
+            "products": self._get_products_data(docs),
+            "taxes": self._get_taxes_data(docs),
+            "tota_summary": self._get_total_summary_data(docs),
+        }
+        return data
+
+    # report data. Auxiliary methods
+    def _get_general_data(self, invoice):
+        type_mapping = {"out_invoice": "FacturaComercial", "out_refund": "FacturaAbono"}
+        return {
+            "Type": type_mapping.get(invoice.move_type),
+            "Ref": invoice.name,
+            "Date": invoice.invoice_date
+            and datetime.strftime(invoice.invoice_date, "%Y-%m-%d"),
+            "Currency": invoice.currency_id.name,
+        }
+
+    def _get_suplier_data(self, invoice):
+        supplier = invoice.company_id.partner_id
+        return {
+            "CIF": supplier._get_voxel_vat(),
+            "Company": supplier.name,
+            "Address": ", ".join(filter(None, [supplier.street, supplier.street2])),
+            "City": supplier.city,
+            "PC": supplier.zip,
+            "Province": supplier.state_id.name,
+            "Country": supplier.country_id.code_alpha3,
+            "Email": supplier.email,
+        }
+
+    def _get_client_data(self, invoice):
+        client = invoice.partner_id
+        return {
+            "SupplierClientID": client.ref,
+            "CIF": client._get_voxel_vat(),
+            "Company": client.commercial_partner_id.name,
+            "Address": ", ".join(filter(None, [client.street, client.street2])),
+            "City": client.city,
+            "PC": client.zip,
+            "Province": client.state_id.name,
+            "Country": client.country_id.code_alpha3,
+            "Email": client.email,
+        }
+
+    def _get_customers_data(self, invoice):
+        return [
+            {
+                "SupplierClientID": invoice.partner_id.ref,
+                "SupplierCustomerID": customer.ref,
+                "Customer": customer.name,
+                "Address": ", ".join(filter(None, [customer.street, customer.street2])),
+                "City": customer.city,
+                "PC": customer.zip,
+                "Province": customer.state_id.name,
+                "Country": customer.country_id.code_alpha3,
+                "Email": customer.email,
+            }
+            for customer in (
+                invoice.mapped("picking_ids.partner_id") or invoice.partner_shipping_id
+            )
+        ]
+
+    def _get_comments_data(self, invoice):
+        return invoice.narration and [{"Msg": invoice.narration}] or []
+
+    def _get_references_data(self, invoice):
+        references = []
+        if invoice.picking_ids:
+            for picking in invoice.picking_ids:
+                picking_date = fields.Datetime.from_string(picking.date)
```

Comment:
I think this line is outdated. The fields of type `date` and `datetime` are now native Python date types, so no conversion is needed.
```suggestion
                picking_date = picking.date
```